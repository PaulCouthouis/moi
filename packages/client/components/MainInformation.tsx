import { MainInformationEntity } from '@frontend/core'
import { x } from '@xstyled/styled-components'

export const MainInformation: React.FC<{ mainInformation: MainInformationEntity }> = (
  { mainInformation: { firstName, lastName, job, description } }
) => {
  return (
    <x.article
      w={{ sm: '65%' }}
      display='flex'
      alignItems='center'
      justifyContent='center'
      p='30px 30px 50px'
      fontFamily='Poppins, Helvetica, sans-serif'
      fontSize={14}
      lineHeight='1.85em'
    >
      <x.div maxWidth='100%'>
        <x.header color='#a5a5a5' fontWeight={300}>
          {job}
        </x.header>
        <x.h1 color='#f5f5f5' fontSize={36} fontWeight={600} lineHeight='1.2em' m='0 0 15px 0'>{firstName} {lastName}</x.h1>
        <x.p color='#d5d5d5' w='65ch' maxWidth='100%'> {description} </x.p>
      </x.div>
    </x.article>
  )
}
