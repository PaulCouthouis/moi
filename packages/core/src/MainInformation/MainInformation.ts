import { Builder } from '../builder.type'

interface MainInformation {
  firstName: string
  lastName: string
  job: string
  description: string
}

const mainInformation = {
  firstName: 'Paul',
  lastName: 'Couthouis',
  job: 'DÃ©veloppeur Front-End',
  description: 'xxx'
}

export const MainInformationBuilder: Builder<MainInformation> = () => mainInformation
