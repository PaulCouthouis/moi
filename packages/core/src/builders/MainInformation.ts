import { Builder, DataBuilder, Repository } from './_builder'

export interface MainInformationEntity {
  firstName: string
  lastName: string
  job: string
  description: string
}

export type MainInformationRepository = Repository<MainInformationEntity>
export const MainInformationBuilder: Builder<MainInformationRepository, MainInformationEntity> = DataBuilder
