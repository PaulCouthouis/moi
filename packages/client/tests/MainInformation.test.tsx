import renderer from 'react-test-renderer'
import { MainInformation } from '../components/MainInformation/MainInformation'

describe('Main Information', () => {
  it('should return information of Paul Couthouis', () => {
    expect(renderer.create(
      <MainInformation
        mainInformation={{
          firstName: 'Paul',
          lastName: 'Couthouis',
          job: 'DÃ©veloppeur Front-End',
          description: 'Lorem elit nostrud sit aute. Velit Lorem incididunt qui consectetur elit sint. Tempor dolore quis et reprehenderit reprehenderit ex cupidatat aliquip duis pariatur dolore aute sit ut.'
        }}
      />)
    ).toMatchSnapshot()
  })
  it('should return information of John Doe', () => {
    expect(renderer.create(
      <MainInformation
        mainInformation={{
          firstName: 'John',
          lastName: 'Doe',
          job: 'Journalist',
          description: 'Incididunt do sint elit officia labore minim Lorem officia qui excepteur. Aute dolor excepteur quis culpa esse veniam amet est aute. Cillum ut amet veniam cillum occaecat duis consectetur irure elit eu. Excepteur anim exercitation do commodo elit et nulla consequat ipsum voluptate proident consequat ullamco do. Eu exercitation sint qui anim.'
        }}
      />)
    ).toMatchSnapshot()
  })
})
