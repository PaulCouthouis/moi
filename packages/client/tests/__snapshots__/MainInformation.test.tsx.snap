// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Information should return information of John Doe 1`] = `
<article
  className="sc-eCApnc gRlSpJ"
>
  <div
    className="sc-cxNHIi ivfhdD"
  >
    <header
      className="sc-efHYUO ernzeX"
    >
      Journalist
    </header>
    <h1
      className="sc-gXfVKN UAOGD"
    >
      John
       
      Doe
    </h1>
    <div
      className="sc-iIgjPs ercSyr"
    >
       
      <p>
        test 2
      </p>
       
    </div>
    <a
      className="sc-bdnxRM cxVKJa"
      href="https://drive.google.com/file/d/1pJKHQ54QYnRS4p8uWf2LaOsALSNy3jFs/view?usp=sharing"
      target="_blank"
    >
      Télécharger mon CV
    </a>
  </div>
</article>
`;

exports[`Main Information should return information of Paul Couthouis 1`] = `
<article
  className="sc-eCApnc gRlSpJ"
>
  <div
    className="sc-cxNHIi ivfhdD"
  >
    <header
      className="sc-efHYUO ernzeX"
    >
      Développeur Front-End
    </header>
    <h1
      className="sc-gXfVKN UAOGD"
    >
      Paul
       
      Couthouis
    </h1>
    <div
      className="sc-iIgjPs ercSyr"
    >
       
      <p>
        test 1
      </p>
       
    </div>
    <a
      className="sc-bdnxRM cxVKJa"
      href="https://drive.google.com/file/d/1pJKHQ54QYnRS4p8uWf2LaOsALSNy3jFs/view?usp=sharing"
      target="_blank"
    >
      Télécharger mon CV
    </a>
  </div>
</article>
`;
